rhapsody/heatmap.py

```diff
@@ -0,0 +1,157 @@
+        'viridis', 'plasma', 'inferno', 'magma', 'cividis']}
+
+
+def ranks_heatmap(ranks, microbe_metadata=None, metabolite_metadata=None,
+                  method='average', metric='euclidean',
+                  color_palette='seismic', margin_palette='cubehelix',
+                  x_labels=False, y_labels=False, level=3, threshold=3):
+    '''
+    Generate clustermap of microbe X metabolite conditional probabilities.
+
+    ranks: pd.DataFrame of conditional probabilities.
+        Microbes (rows) X metabolites (columns).
+    microbe_metadata: pd.Series of microbe metadata for annotating plots
+    metabolite_metadata: pd.Series of metabolite metadata for annotating plots
+    method: str
+        Hierarchical clustering method used in clustermap.
+    metric: str
+        Hierarchical clustering distance metric used in clustermap.
+    color_palette: str
+        Color palette for clustermap.
+    margin_palette: str
+        Name of color palette to use for annotating metadata
+        along margin(s) of clustermap.
+    x_labels: bool
+        Plot x-axis (metabolite) labels?
+    y_labels: bool
+        Plot y-axis (microbe) labels?
+    level: int
+        taxonomic level for annotating clustermap. Set to -1 if not parsing
+        semicolon-delimited taxonomies or wish to print entire annotation.
+    threshold: int
+        Minimum absolute value of conditional probabilities to plot.
+        metabolites/taxa that do not have at least one CP above the
+        threshold will be removed.
+
+    Returns clustermap
+    '''
+    # filter microbes
+    abs_vals = ranks.abs()
+    ranks = ranks[abs_vals.max(axis=1) > threshold]
+    # filter metabolites
+    ranks = ranks[ranks.columns[(abs_vals > threshold).any()]]
+
+    # subset microbe metadata based on rows/columns
+    if microbe_metadata is not None:
+        microbe_metadata = microbe_metadata.reindex(ranks.index)
+        # parse semicolon-delimited taxonomy
+        if level > -1:
+            microbe_metadata = _parse_taxonomy_strings(microbe_metadata, level)
+        # map metadata categories to row colors
+        row_colors, row_class_colors = _parse_heatmap_metadata_annotations(
+            microbe_metadata, margin_palette)
+    else:
+        row_colors = None
+
+    # subset metabolite metadata based on rows/columns
+    if metabolite_metadata is not None:
+        metabolite_metadata = metabolite_metadata.reindex(ranks.columns)
+        # map metadata categories to column colors
+        col_colors, col_class_colors = _parse_heatmap_metadata_annotations(
+            metabolite_metadata, margin_palette)
+    else:
+        col_colors = None
+
+    # Generate heatmap
+    hotmap = sns.clustermap(ranks, cmap=color_palette, center=0,
+                            col_colors=col_colors, row_colors=row_colors,
+                            figsize=(12, 12), method=method, metric=metric,
+                            cbar_kws={'label': 'Conditional\nProbability'})
```

Comment:
`cbar_kws={'label': 'Conditional\nProbability'})` -> `cbar_kws={'label': 'Log Conditional\nProbability'})`

Reply:
I am adding this in the changes I will push up soon, but I am conflicted. Will the `Conditionals` always contain log conditional probabilities? Since currently the answer is yes, I have made this change, but we may want to consider if you foresee other types of data being contained in the `Conditionals`.